### PR TITLE
Validate new public key in EthAccount

### LINKED ISSUE #818: Validate new public key in Account public key setter functions
To minimize the risk of changing the key to a wrong one, there could be an impl in `AccountComponent` that validates ownership e.g. by requiring a signed message. It might make sense to standardize such message.

### PR DESCRIPTION
**Pull request by ericnordelo**
Partially fixes #818  

#### PR Checklist

- [x] Tests
- [x] Documentation
- [x] Added entry to CHANGELOG.md 
- [ ] Tried the feature on a public network

### CODE REVIEW

File: src/tests/account/test_eth_account.cairo [hunk 1]
```diff
@@ -583,6 +610,32 @@ fn test__set_public_key() {
...
 //
 
+fn get_accept_ownership_signature() -> Span<felt252> {
+    let mut output = array![];
+
+    // 0x02357636c7266194923ea0b270bd563853c1fdea875b3cc6567f6d08cbccc3be =
+    // PoseidonTrait::new()
+    //             .update_with('StarkNet Message')
+    //             .update_with('accept_ownership')
+    //             .update_with(ACCOUNT_ADDRESS())
```
Comment (andrew-fleming) [inline src/tests/account/test_eth_account.cairo:620]: Isn't it weird that `ACCOUNT_ADDRESS` is `0`? I know it's just for tests and AFAICT it doesn't sully the integrity of the tests we have, but IMO this should be non-zero. Do you agree?

File: docs/modules/ROOT/pages/api/account.adoc [hunk 2]
```diff
@@ -459,6 +478,19 @@ Emits an {OwnerAdded} event.
 
 Validates that the caller is the account itself. Otherwise it reverts.
 
+[.contract-item]
+[[EthAccountComponent-assert_valid_new_owner]]
+==== `[.contract-item-name]#++assert_valid_new_owner++#++(self: @ComponentState, current_owner: EthPublicKey, new_owner: EthPublicKey, signature: Span<felt252>)++` [.item-kind]#internal#
+
+Validates that new owner accepted the ownership of the contract through a signature.
```
Comment (andrew-fleming) [inline docs/modules/ROOT/pages/api/account.adoc:485]:
```suggestion
Validates that `new_owner` accepted the ownership of the contract through a signature.
```

**Review by andrew-fleming — APPROVED** [review]
I left a couple tiny suggestions on styling and a question regarding tests. Otherwise, this looks good to me!

File: src/tests/account/test_eth_account.cairo (hunk 1 above)
Comment (ericnordelo) [reply src/tests/account/test_eth_account.cairo:620]: Good catch, and I agree.

Comment (ericnordelo) [reply src/tests/account/test_eth_account.cairo:620]: Fixed!

**Pull request merged**